timemachine/fe/utils.py

```diff
@@ -225,6 +227,80 @@ def plot_atom_mapping_grid(mol_a, mol_b, core, num_rotations=5):
     )
 
 
+def view_atom_mapping_3d(
+    mol_a,
+    mol_b,
+    core=None,
+    colors=plt.rcParams["axes.prop_cycle"].by_key()["color"],
+    show_idx_labels=False,
+    width=800,
+    height=500,
+):
+    """Produce a 3D rotatable view of a pair of molecules using py3Dmol. If `core` is specified, displays a second row
+    where the atoms are colored according to the atom mapping.
+
+    Parameters
+    ----------
+    mol_a, mol_b : rdkit mols
+        Input mols
+
+    core : ndarray or list of list of int or None, optional
+        Atom mapping. If passed, display a second row with atoms color-coded according to mapping
+
+    colors : list of str, optional
+        Colors to use for highlighting atoms by mapping
+
+    show_idx_labels : bool, optional
+        Whether to display atom indices
+
+    width, height : int
+        Dimensions of the view
+
+    Returns
+    -------
+    py3Dmol.view
+    """
+
+    try:
+        import py3Dmol
+    except ImportError:
+        raise ValueError("requires py3Dmol to be installed")
```

Comment:
nit: ValueError -> ImportError?

Reply:
I'm unclear on a good convention for handling missing optional dependencies, but reraising an `ImportError` didn't seem quite right to me. Changed to raise a `RuntimeError` chained with the original `ImportError` in https://github.com/proteneer/timemachine/pull/971/commits/0df7735f6635814d0c8ae5af879c7596e925f240. Does this seem OK to you?

Reply:
Looks good!

(Any of these options seems fine to me: ValueError, ImportError, or RuntimeError. I don't know of a convention. Elsewhere in timemachine, import errors from missing optional dependencies are not explicitly handled, I think -- see e.g. https://github.com/proteneer/timemachine/blob/f70586d95af0231b8a663749c268e440457f0d8c/timemachine/ff/handlers/nonbonded.py#L32-L33  . In any case, the required action to fix the error will be clear!)